src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -922,6 +922,24 @@ public ImmutableArray<T> Sort()
         }
 
         /// <summary>
+        /// Sorts the elements in the entire <see cref="ImmutableArray{T}"/> using
+        /// the specified <see cref="Comparison{T}"/>.
+        /// </summary>
+        /// <param name="comparison">
+        /// The <see cref="Comparison{T}"/> to use when comparing elements.
+        /// </param>
+        /// <returns>The sorted list.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="comparison"/> is null.</exception>
+        [Pure]
+        public ImmutableArray<T> Sort(Comparison<T> comparison)
+        {
+            Requires.NotNull(comparison, "comparison");
+
+            var self = this;
+            return self.Sort(Comparer<T>.Create(comparison));
```

Comment:
True. I considered the alternative, but looking at the Sort(IComparer) method, it's non-trivial to replicate and then we have twice the maintenance burden. I ended with deciding that adding the API to enable customers is enabled by this smaller change, and we can decide to pay the cost of doubling the implementation if we later decide saving the allocation is worthwhile.